src/api_server/src/lib.rs

```diff
@@ -129,6 +136,12 @@ impl ApiServer {
         loop {
             match server.requests() {
                 Ok(request_vec) => {
+                    if let Some(exit_code) = *self.encountered_fatal_error.borrow() {
```

Comment:
This needs to happen inside the loop below; right after sending the HTTP response (after `server.respond()`).

Ideally it should also flush the http socket to make sure the response is sent before exiting the process.

Reply:
While I agree to this approach, the function `server.requests()` also does the job of responding to the client when there are `epoll::EventSet::OUT` events to be processed. Do you think it would be feasible to call again `server.requests()` before exiting the process, so that it could flush the current response queue?

Reply:
After discussing with @acatangiu , who guided me to reach out to @georgepisaltu to dive deeper into the case, we have decided on the following: the verification for encountering fatal errors will be done after the server response at line 148. If we have a fatal error, a new HttpServer function will check if we have pending outgoing writes in a loop. If we do, we call `server.requests()` to address the current response queue, while also returning a 500 Internal Server Error to any new incoming requests. We kill the process afterwards.